apps/examples/ble_rmc/ble_rmc_main.c

```diff
@@ -192,6 +190,110 @@ static uint8_t g_adv_resp[] = {
+};
+
+static ble_client_callback_list client_config = {
+	ble_device_disconnected_cb,
+	ble_device_connected_cb,
+	ble_operation_notification_cb,
+};
+
+static ble_server_init_config server_config = {
+	ble_server_connected_cb,
+	true,
+	gatt_profile, sizeof(gatt_profile) / sizeof(ble_server_gatt_t)};
+
+static ble_result_e ble_connect_common(bool is_auto)
+{
+	ble_result_e ret = BLE_MANAGER_FAIL;
+	struct timespec abstime;
+	ble_client_state_e cli_state = BLE_CLIENT_NONE;
+	ble_conn_info conn_info = { 0, };
+
+	memcpy(conn_info.addr.mac, g_target.mac, BLE_BD_ADDR_MAX_LEN);
+	RMC_LOG(RMC_CLIENT_TAG, "BLE mac type : %d\n", g_target.type);
+	conn_info.addr.type = g_target.type;
+	conn_info.conn_interval = 8;
+	conn_info.slave_latency = 128;
+	conn_info.mtu = 240;
+	conn_info.scan_timeout = 1000;
+	conn_info.is_secured_connect = true;
+
+	sem_init(&g_conn_sem, 0, 0);
+
+	if (g_ctx == NULL) {
+		g_ctx = ble_client_create_ctx(&client_config);
+	}
+	if (g_ctx == NULL) {
+		RMC_LOG(RMC_CLIENT_TAG, "create ctx fail\n");
+		return 0;
+	} else {
+		RMC_LOG(RMC_CLIENT_TAG, "create ctx ok[%p]\n", g_ctx);
+	}
+
+	ret = ble_client_autoconnect(g_ctx, is_auto);
+	if (ret != BLE_MANAGER_SUCCESS) {
+		RMC_LOG(RMC_CLIENT_TAG, "fail to set autoconnect=%d [%d]\n", is_auto, ret);
+	}
+
+	cli_state = ble_client_get_state(g_ctx);
+	RMC_LOG(RMC_CLIENT_TAG, "Client State [ %s ]\n", __client_state_str(cli_state));
+
+	ret = ble_client_connect(g_ctx, &conn_info);
+	if (ret != BLE_MANAGER_SUCCESS) {
+		RMC_LOG(RMC_CLIENT_TAG, "connect fail[%d]\n", ret);
+	}
+
+	cli_state = ble_client_get_state(g_ctx);
+	RMC_LOG(RMC_CLIENT_TAG, "Client State [ %s ]\n", __client_state_str(cli_state));
+
+	ret = clock_gettime(CLOCK_REALTIME, &abstime);
+
+	abstime.tv_sec = abstime.tv_sec + 10;
+	abstime.tv_nsec = 0;
+
+	int status = sem_timedwait(&g_conn_sem, &abstime);
+	sem_destroy(&g_conn_sem);
+	int errcode = errno;
+	if (status < 0) {
+		RMC_LOG(RMC_CLIENT_TAG, "ERROR: sem_timedwait failed with: %d\n", errcode);
+		return 0;
+	} else {
+		RMC_LOG(RMC_CLIENT_TAG, "PASS: sem_timedwait succeeded\n");
+	}
+
+	cli_state = ble_client_get_state(g_ctx);
+	RMC_LOG(RMC_CLIENT_TAG, "Client State [ %s ]\n", __client_state_str(cli_state));
+
+	if (cli_state != BLE_CLIENT_CONNECTED) {
+		RMC_LOG(RMC_CLIENT_TAG, "BLE is not connected");
+		return 0;
+	} else {
+		RMC_LOG(RMC_CLIENT_TAG, "BLE is connected\n");
+	}
+	
+	ble_attr_handle attr_handle;
+
+	attr_handle = 0xff03;
+	ret = ble_client_operation_enable_notification(g_ctx, attr_handle);
+	if (ret != BLE_MANAGER_SUCCESS) {
+		RMC_LOG(RMC_CLIENT_TAG, "Fail to enable noti handle1[%d]\n", ret);
+	} else {
+		RMC_LOG(RMC_CLIENT_TAG, "Success to enable noti handle1.\n");
+	}
+
+	attr_handle = 0x006e;
+	ret = ble_client_operation_enable_notification(g_ctx, attr_handle);
+	if (ret != BLE_MANAGER_SUCCESS) {
+		RMC_LOG(RMC_CLIENT_TAG, "Fail to enable noti handle2[%d]\n", ret);
+	} else {
+		RMC_LOG(RMC_CLIENT_TAG, "Success to enable noti handle2.\n");
+	}
```

Comment:
Returning result is missed. Please add `return ret` when exit.

Reply:
Ok, I will fix it